### ISSUE TITLE
Purchased Plugins page can throw exception if purchased plugin not found

### ISSUE THREAD

**Issue by cabal95** [SparkDevNetwork/Rock#2882]
### Prerequisites
* [X] Put an X between the brackets on this line if you have done all of the following:
    * Can you reproduce the problem on a fresh install or the [demo site](http://rock.rocksolidchurchdemo.com/)?
    * Did you include your Rock version number and [client culture](https://github.com/SparkDevNetwork/Rock/wiki/Environment-and-Diagnostics-Information) setting?
    * Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3ASparkDevNetwork+-repo%3ASparkDevNetwork%2FSlack) to see if your bug or enhancement is already reported?

### Description

If you have purchased a plugin but it is not installed on your Rock server (for example, you have multiple servers) then going to the Purchased Plugins page will throw a null reference exception. This is because the block currently assumes the plugin is installed instead of checking for a not-installed possibility.

Can work on a PR for this myself.

### Steps to Reproduce

1. Install 2 plugins on production.
2. Replicate server to a dev box.
3. On production install a third plugin.
4. On dev box browse to the Purchased Plugins page.

**Expected behavior:** 

Page should show a "not installed, install now" button, similar to how it does if there is an update available.

**Actual behavior:** 

![image](https://user-images.githubusercontent.com/1119863/37543480-fd7afcf4-291e-11e8-8ab4-b203b3b62955.png)

### Versions

* **Rock Version:** 7.3, 8.0
* **Client Culture Setting:** en-US

**Comment by edmistj** [comment]
I thought this resolved it.
https://github.com/SparkDevNetwork/Rock/commit/6855f662756c7efd670b093a148f90938e014cd9

Is there another issue?

**Comment by cabal95** [comment]
Yes. Apparently `installedPackage` can **also** be null if your Org has purchased the package but it is not installed on this server (as in the example of installing it on production and then going to view this page on dev). I have a 6-line PR for this ready.

**Comment by mikejed** [comment]
I still get an exception on my v7.3 dev instance's Purchases page, because there are plugins which we purchased (and installed only on the production server) which are not installed on the dev server. That's the issue which Daniel is looking into here.

**Comment by edmistj** [comment]
Please note that this should be applied to the hotfix-v74 branch in your PR.
